cmd/cli/exec/exec.go

```diff
@@ -120,6 +121,14 @@ func PrepareJob(cmd *cobra.Command, cmdArgs []string, unknownArgs []string, opti
 	// don't have a template, then we don't know how to submit that job type.
 	jobType = cmdArgs[0]
 
+	for i := range cmdArgs {
+		// If any parameters were quoted, we should make sure we try and add
+		// them back in after they were stripped for us.
+		if strings.Contains(cmdArgs[i], " ") {
+			cmdArgs[i] = fmt.Sprintf(`"%s"`, cmdArgs[i])
+		}
+	}
+
```

Comment:
What if `cmdArgs[...]` was `'hello "world"!'` i.e. single quotes.

I feel like trying to handle shell escaping is going to be painful. Can we just ensure these parameters are passed to the program directly without any further shell interpretation?

Reply:
Yeah, I'll take another look, I think it was giving me problems at the Python end but will double-check

Reply:
We do need to quote these strings, as they are used as args later. e.g.

```shell
python -c "import this"
```

needs to translate to 

```shell
launcher.py -- -c "import this"
```

Rather than DIY I've used `shellescape` which is based on the Python version of the same name.